jgrapht-core/src/test/java/org/jgrapht/alg/VertexCoverTest.java

```diff
@@ -0,0 +1,264 @@
+        assertTrue(isCover(g1, vertexCover));
+        assertEquals(vertexCover.getWeight(), 5.0);
+    }
+
+    /**
+     * Cubic graph with 8 vertices (Optimal=7)
+     */
+    public void testCubic(){
+        UndirectedGraph<Integer, DefaultEdge> g1=new SimpleGraph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g1, Arrays.asList(0,1,2,3,4,5,6,7,8,9,10,11));
+        g1.addEdge(0,1);
+        g1.addEdge(0,9);
+        g1.addEdge(0,7);
+        g1.addEdge(1, 2);
+        g1.addEdge(1,5);
+        g1.addEdge(2,3);
+        g1.addEdge(2,4);
+        g1.addEdge(3,4);
+        g1.addEdge(3,5);
+        g1.addEdge(4,11);
+        g1.addEdge(5,6);
+        g1.addEdge(6,7);
+        g1.addEdge(6,8);
+        g1.addEdge(7,8);
+        g1.addEdge(8,10);
+        g1.addEdge(9,10);
+        g1.addEdge(9,11);
+        g1.addEdge(10,11);
+        MinimumVertexCoverAlgorithm<Integer, DefaultEdge> mvc1=new RecursiveExactVCImpl<>();
+        VertexCover<Integer> vertexCover=mvc1.getVertexCover(g1);
+        assertTrue(isCover(g1, vertexCover));
+        assertEquals(vertexCover.getWeight(), 7.0);
+    }
+
+    /**
+     * Graph with 6 vertices in the shape >-< (Optimal=2)
+     */
+    public void testWhisker(){
+        UndirectedGraph<Integer, DefaultEdge> g1=new SimpleGraph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g1, Arrays.asList(0,1,2,3,4,5));
+        g1.addEdge(0,2);
+        g1.addEdge(1,2);
+        g1.addEdge(2,3);
+        g1.addEdge(3,4);
+        g1.addEdge(3,5);
+        MinimumVertexCoverAlgorithm<Integer, DefaultEdge> mvc1=new RecursiveExactVCImpl<>();
+        VertexCover<Integer> vertexCover=mvc1.getVertexCover(g1);
+        assertTrue(isCover(g1, vertexCover));
+        assertEquals(vertexCover.getWeight(), 2.0);
+    }
+
+    // ------- Helper methods ------
+
+    /**
+     * Checks if the specified vertex set covers every edge of the graph. Uses
+     * the definition of Vertex Cover - removes every edge that is incident on a
+     * vertex in vertexSet. If no edges are left, vertexSet is a vertex cover
+     * for the specified graph.
+     *
+     * @param vertexCover the vertex cover to be tested for covering the graph.
+     * @param g the graph to be covered.
+     *
+     * @return returns true if the provided vertex cover is a valid cover in the given graph
+     */
+    protected boolean isCover(
+        Graph<Integer, DefaultEdge> g,
+        VertexCover<Integer> vertexCover)
+    {
+        Set<DefaultEdge> uncoveredEdges = new HashSet<>(g.edgeSet());
+        for (Integer v : vertexCover.getVertices())
+            uncoveredEdges.removeAll(g.edgesOf(v));
+
+        return uncoveredEdges.isEmpty();
+    }
+
+    /**
+     * Create a random PSEUDO graph of TEST_GRAPH_SIZE nodes.
+     *
+     * @return
+     */
+    protected Graph<Integer, DefaultEdge> createRandomPseudoGraph()
+    {
+        // TODO: move random graph generator to be under GraphGenerator
```

Comment:
This is a remnant left by the previous author. I'll see whether I can swap it out for the one in the GraphGenerator package.